### PR TITLE
Refactor L#

### PR DESCRIPTION
**Pull request by cbeauchesne**
This PR fix a small bug, but it's not the main objective of it.

This code must be as clear as possible, and anybody must be able to understand it, and if neccessary, fix something. 

I've changed the strucutre from a three-steps (get blocks/build tree/render html) process, to a two step process : 

1. Consider L# as a simple table syntax, and render it
2. Post process HTML in order to replace any L# by coresponding numbering

So, @asaunier, @olaurendeau, @brunobesson, would it be possible for you to take some times (maybe 30mn?) to read it, and see this points : 

1. Are the comments clear enough?
2. The code is understandable?
3. The structure is not too complex?

Thank you in advance !

### CODE REVIEW

File: c2corg_ui/format/ltag.py [hunk 1]
```diff
@@ -8,172 +8,149 @@
...
-        self.supported = None
+        self.supported = True
 
         # If no relative pattern is present, then label are allowed
         # As now, the only relative pattern handled is a simple L#
-        self.allow_labels = None
+        self.allow_labels = True
 
         # if numbering contains a label, then relatives patterns
         # are no more allowed
```
Comment (brunobesson) [inline c2corg_ui/format/ltag.py]: not allowed anymore

File: c2corg_ui/format/ltag.py [hunk 2]
```diff
@@ -8,172 +8,149 @@
...
 
         # Values for relative patterns
-        self.value = None
+        self.value = {"R": 0, "L": 0}
 
         # One way switch
-        self.supported = None
+        self.supported = True
 
         # If no relative pattern is present, then label are allowed
```
Comment (brunobesson) [inline c2corg_ui/format/ltag.py]: labels

Comment (cbeauchesne) [reply c2corg_ui/format/ltag.py]: ✅

File: c2corg_ui/format/ltag.py (hunk 1 above)
Comment (cbeauchesne) [reply c2corg_ui/format/ltag.py]: ✅

**Comment by cbeauchesne** [comment]
C'est pénible, le script de check de syntax m'envoie peter sur un fichier que j'ai pas modifié. Je corrige, et maintenant il y a un conflit.

C'est usant.

**Comment by cbeauchesne** [comment]
I will try tonight. Was in a bad mood yesterday...

**Comment by brunobesson** [comment]
I'll be available on Friday evening if needed.

**Comment by cbeauchesne** [comment]
Well, the last commit for L#, made for V6.

@olaurendeau @brunobesson @asaunier , could you have a quick look on it?

It handle allowed V6 syntax (see [1](https://www.camptocamp.org/articles/305462/fr/aide-topoguide-l-balise-de-description-des-longueurs) and [2](https://forum.camptocamp.org/t/305462-fr/196615/11?u=charlesb) and no more.

Table parsing and numbering stuff are separated, and it should not be too complex to understand.

**Comment by cbeauchesne** [comment]
Up guys. Weather is awfull this WE, best occasion ever to make C2C better than ever. 😉

More seriously, it's quite important that this code is understandable by anybody, as we do not want to live L# nightmare again. If I'm the only one who can read my code, it will be a fail. It should take 15 minutes. If it'd more, then I need to comment it more.

**Comment by brunobesson** [comment]
Maybe time this evening but notre sure. Next weekend otherwise.

**Comment by cbeauchesne** [comment]
Thanks!

**Pull request merged**